pkg/controller/actions/deploy/action_deploy.go

```diff
@@ -469,6 +469,14 @@ func (a *Action) apply(
 		if found {
 			unstructured.RemoveNestedField(obj.Object, "rules")
 		}
+	// For observability components, we preserve user changes to resource requests/limits
+	case gvk.MonitoringStack, gvk.TempoStack, gvk.TempoMonolithic, gvk.OpenTelemetryCollector:
+		if old == nil || resources.GetAnnotation(old, annotations.ManagedByODHOperator) == "true" {
+			break
+		}
+		if err := MergeObservabilityResources(old, obj); err != nil {
+			return nil, fmt.Errorf("failed to merge %s %s/%s: %w", obj.GetKind(), obj.GetNamespace(), obj.GetName(), err)
+		}
```

Comment:
Just thinking, correct me if it's wrong assumption. After first deployment of MonitoringStack we fill in `spec.resource` stanza. Then whatever metrics resources configuration is subsequently set in DSCI will not be taken into account, right? Because `MergeObservabilityResources` will always take resources part from the already deployed object. So this will ignore whatever is coming from DSCI.

For other resource it will work because we have dedicated Tempo/Collector keys for those resources and they are not set using DSCI unlike MonitoringStack.

Reply:
We should remove the fields from the DSCI.